File: benchmark/nodejs/hanoi.js
function hanoi(n) {
    if (n === 1) {
        return 1;
    }
    return hanoi(n - 1) + 1 + hanoi(n - 1);
}
const start = Date.now();
const moves = hanoi(20);
const end = Date.now();
console.log(`Moves: ${moves}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/ruby/hanoi.rb
def hanoi(n)
  return 1 if n == 1
  hanoi(n - 1) + 1 + hanoi(n - 1)
end
start = Time.now.to_f * 1000
moves = hanoi(20)
finish = Time.now.to_f * 1000
puts "Moves: #{moves}"
puts "Time: #{(finish - start).round}ms"

